flink-runtime/src/main/java/org/apache/flink/runtime/io/network/NetworkEnvironment.java

```diff
@@ -264,11 +264,15 @@ public void setupInputGate(SingleInputGate gate) throws IOException {
 		int maxNumberOfMemorySegments;
 		try {
 			if (enableCreditBased) {
+				Preconditions.checkArgument(networkBuffersPerChannel > 0,
+					"In credit-based flow control mode, networkBuffersPerChannel must be configured greater than 0.");
 				maxNumberOfMemorySegments = gate.getConsumedPartitionType().isBounded() ?
-					extraNetworkBuffersPerGate : Integer.MAX_VALUE;
+					gate.getNumberOfInputChannels() * (networkBuffersPerChannel - 1) +
```

Comment:
I think we should not change the setting behaviors internally. Otherwise the users might not be aware of this. 

The motivation of this ticket is decreasing the default value for per-channel config. But if users want to config this value explicitly, we should obey it.